lib/bolt/inventory/inventory2.rb

```diff
@@ -199,6 +212,41 @@ def expand_targets(targets)
       end
       private :expand_targets
 
+      def resolve_plugins(plugins, data)
+        if data.is_a? Hash
+          data.each_with_object({}) do |(k, v), acc|
+            v = case k
+                when 'config'
+                  config_only_plugin(plugins, v)
+                when 'target-lookups'
```

Comment:
Anything under the `config` key can be safely converted to concurrent delay *unless* it is nested under a `target-lookups` key. The `config_only_plugin` method should only be called on the value pointed to by a `config` key that is not nested under a `target-lookups` key.